core/src/gl/framebuffer.cpp

```diff
@@ -72,6 +72,21 @@ GLuint FrameBuffer::readAt(float _normalizedX, float _normalizedY) const {
     return pixel;
 }
 
+FrameBuffer::PixelRect FrameBuffer::readRect(float _normalizedX, float _normalizedY, float _normalizedW, float _normalizedH) const {
+
+    PixelRect rect;
+    rect.left = floorf(_normalizedX * m_width);
+    rect.bottom = floorf(_normalizedY * m_height);
+    rect.width = ceilf(_normalizedW * m_width);
+    rect.height = ceilf(_normalizedH * m_height);
+
+    rect.pixels.resize(rect.width * rect.height);
+
+    GL::readPixels(rect.left, rect.bottom, rect.width, rect.height, GL_RGBA, GL_UNSIGNED_BYTE, rect.pixels.data());
```

Comment:
From this: http://docs.gl/es2/glReadPixels it sounds like those ranges describe the positions of the returned pixels, where `x`, `y`, `width`, and `height` are the arguments to `glReadPixels`, not the dimensions of the framebuffer itself, so I think we're ok here.

In the cases where `rect.left == m_width` or `rect.bottom == m_height` we would end up sampling with 0 width or height, so there shouldn't be any undefined results.